pkg/provider/vault/vault.go

```diff
@@ -158,6 +159,30 @@ type VClient struct {
 	addHeader    func(key, value string)
 }
 
+func ClientWithLoginMock(c *vault.Config) (Client, error) {
```

Comment:
It feels weird to have this in a non-test file. Can you move this to `vault/fake` ?

Reply:
Hi @moolen, I'm having trouble isolating this into `vault/fake`; it depends on the `Client` interface and `VClient` struct from the main `vault` package, so it would probably require creating some new package (e.g., `vault/util`) to hold this shared code, or duplicating some of the code in the `generator/vault` tests.

Can you suggest the correct approach, please? Thanks 🙂

Reply:
Both options are fine, i think i'd lean towards a package for it.